Add author and editor lists

GreenXenith opened the pull request:
`authors` and `editors` are arrays. Posts should ideally always have at least an `authors` array.

![image](https://github.com/minetest/blog/assets/24834740/f3a3db53-6b31-4e03-baae-aeecaadf35c7)

rubenwardy commented:
Probably worth amending articles to use this

rubenwardy reviewed (changes requested):
I think you need to update the posts to use `authors` rather than `author` otherwise existing posts will lose authorship info

Unless Jekyll does this automatically from `authors`

GreenXenith commented:
@rubenwardy Posts are now updated with these fields.

I've opted to exclude listed authors/editors from the contributors list, and I added PR reviewers with minor changes to the contributors list.

The formatting is also a bit nicer now (`Written by Name1`, `Written by Name1 and Name2`, `Written by Name1, Name2, and Name3`), and applied to the contributors list.

rubenwardy reviewed (approved):
I've not tested this but the changes look good to me

rubenwardy commented:
Will need to update the 5.11.0 release post to use authors
